plugin/pkg/admission/taint/admission.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	// Allow admin users or request over the insecure socket to do whatever they want
+	if tpmAdmin == true {
+		glog.Errorf("TPMadmin = true")
+		return nil
+	}
+
+	// Unprivileged users aren't allowed to perform actions that would alter the trust state
+	switch kind {
+	case api.Kind("ConfigMap"):
+		name := a.GetName()
+		// A create operation may not provide a name directly, so pull it out of the object if we didn't get one
+		if name == "" {
+			configmap, ok := a.GetObject().(*api.ConfigMap)
+			if !ok {
+				return apierrors.NewBadRequest("Resource was marked with kind ConfigMap but couldn't be type converted")
+			}
+			name = configmap.Name
+		}
+		// Only permit TPM admins to modify these, otherwise nodes can circumvent security policy
+		if name == "taint.coreos.com" || name == "tpm.coreos.com" {
+				return apierrors.NewBadRequest("Unauthorised attempt to modify taint configuration")
+		}
+		return nil
+	case api.Kind("Node"):
+		operation := a.GetOperation()
+		if operation != admission.Create && operation != admission.Update {
+			return nil
+		}
+
+		node, ok := a.GetObject().(*api.Node)
+		if !ok {
+			return apierrors.NewBadRequest("Resource was marked with kind Node but couldn't be type converted")
+		}
+
+		if operation == admission.Create {
+			err := invalidateNode(node)
+			if err != nil {
+				return fmt.Errorf("Unable to invalidate node: %v", err)
+			}
+			return nil
+		}
+
+		// If an external update tries to switch a node from untrusted to trusted, force it back to untrusted
+		untrusted, err := isUntrusted(node)
+		if err != nil {
+			return fmt.Errorf("Unable to identify node trusted status: %v", err)
+		}
+		if untrusted == false {
+			oldNode, err := t.client.Core().Nodes().Get(node.Name)
+			if err != nil {
+				return fmt.Errorf("Attempting to update a node that doesn't exist? %v", err)
+			}
+			oldUntrusted, err := isUntrusted(oldNode)
+			if err != nil || oldUntrusted {
+				glog.Errorf("User %v attempted to flag untrusted node %v as trusted", user, node.Name)
+				invalidateNode(node)
+			}
+		}
+	case api.Kind("Pod"):
+		operation := a.GetOperation()
+		if operation != admission.Update {
+			return nil
+		}
+		pod, ok := a.GetObject().(*api.Pod)
+		if !ok {
+			return apierrors.NewBadRequest("Resource was marked with kind Pod but couldn't be type converted")
+		}
+		tolerates, err := toleratesUntrusted(pod)
+		if err != nil {
+			return fmt.Errorf("Unable to identify pod toleration status: %v", err)
+		}
+		if tolerates == true {
+			oldPod, err := t.client.Core().Pods(namespace).Get(pod.Name)
+			if err != nil {
+				return fmt.Errorf("Attempting to update a pod that doesn't exist? %v", err)
+			}
+			oldTolerates, err := toleratesUntrusted(oldPod)
+			if err != nil || oldTolerates == false {
+				glog.Errorf("User %v attempted to flag pod %v as tolerating untrusted nodes", user, pod.Name)
+				return apierrors.NewBadRequest("Invalid attempt to declare that a pod tolerates untrusted nodes")
+			}
+		}
```

Comment:
Yes. The idea is to prevent someone reconfiguring a pod that doesn't tolerate untrusted nodes into one that does.